### DISCUSSION TITLE
Raise an event in case of Task not found

### DISCUSSION THREAD

**Discussion by mithlesh135** [Netflix/conductor#3470] [Ideas]
Currently I have seen conductor just logs and returns in case task is not found in redis while executing an async task, can we raise an event in such case so that cleanup or some action can be taken for people who have deployed conductor in their env.

**Comment by v1r3n** [comment]
Hi @mithlesh135 what version of Conductor are you using?

**Reply by mithlesh135** [reply]
Hi @v1r3n 

We are on 3.8.0

**Reply by mithlesh135** [reply]
Hi @v1r3n 

Is it fixed in recent versions of conductor ?

**Reply by manan164** [reply]
Hi @mithlesh135 , Under what condition do you see that the conductor is not able to find the task? Ideally, all tasks should be in redis.

**Reply by mithlesh135** [reply]
Hi @manan164 

It happens on heave load on conductor, if we give 10K execution concurrently, for some of them it fails with this message, but tasks do remain in queues and get retried and fail again, taking system in always trying to execute those tasks mode.

**Reply by mithlesh135** [reply]
Hi @manan164 @v1r3n 

Any update on this, we are still seeing this happening, can we get a handle that we can use for cleanup.

**Reply by manan164** [reply]
Hi @mithlesh135 , we have fixed this. Our fix does not contain raising an event when a task is not found but we are eliminating the scenarios where we get the task not found error. It is currently in a lower environment. We will merge this in few days.

**Reply by mithlesh135** [reply]
Thanks @manan164, also if you could share the PR that I can follow that will help, as once it will be finalized I'll update our code base

**Reply by mithlesh135** [reply]
Hi @manan164 

Could you let me know in which release fix will be available, does 3.13.5 has the fix for this ?

**Reply by mithlesh135** [reply]
Hi @manan164 / @v1r3n 

could you let me know if this fix is present in 3.13.5 so that we can upgrade our library

**Reply by mithlesh135** [reply]
Hi @manan164 / @v1r3n

Any update on this, is this fixed in latest release.

**Reply by manan164** [reply]
Hi @mithlesh135 , Can you please join the community? wan chat [there](https://join.slack.com/t/orkes-conductor/shared_invite/zt-xyxqyseb-YZ3hwwAgHJH97bsrYRnSZg) for more realtime collaboration. We are working on a fix to remove the message from the queue when it is not found but I would like to understand the scenario where this occurs.

**Reply by mithlesh135** [reply]
Sure
